### ISSUE TITLE
ROL test timing out on ATDM intel-18 mpich build

### ISSUE THREAD

**Issue by fryeguy52** [trilinos/Trilinos#4251]
CC: @trilinos/rol, @rppawlo (Trilinos Nonlinear Solvers Product Lead), @bartlettroscoe, @fryeguy52 
 
## Description
 
As shown in [this query](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-10-09&filtercount=4&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt&field2=testname&compare2=61&value2=ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4&field3=buildstarttime&compare3=84&value3=2019-01-23&field4=buildstarttime&compare4=83&value4=2019-01-01) the test:
* ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4

is timing out in the build:
* Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt

## Current Status on CDash

The current status of this test in this build for the current testing day can be found [here](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-10-09&filtercount=4&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt&field2=testname&compare2=61&value2=ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4&field3=buildstarttime&compare3=84&value3=today&field4=buildstarttime&compare4=83&value4=yesterday)

## Steps to Reproduce

One should be able to reproduce this failure on a machine with a cee rhel6 environment as described in:
* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

More specifically, the commands given for a machine with a cee rhel6 environment are provided at:
* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#cee-rhel6-environment

The exact commands to reproduce this issue should be:
```
$ cd <some_build_dir>/
$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt
$ cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_Rol=ON \
 $TRILINOS_DIR
$ make NP=16
$ ctest -j16
```

**Comment by dridzal** [comment]
@fryeguy52 @trilinos/rol 
This test runs in **8 seconds** on my machine with SEMS rhel6_gcc-6.1.0.  Is there a different ATDM build that we could compare to?  I suspect an Intel compiler issue.

**Comment by dridzal** [comment]
@jmgate @bartlettroscoe 
There are many tests in ROL that take 10-20 times longer when I run
  `  ctest -jN`
where N is pretty much any number larger than 1 (2, 4, 8, 16), compared to simply running
  `  ctest`
I have no idea how to "fix" this.  What I find very strange is that there seems to be an attempt to count MPI ranks for each batch of tests, but that results in bad performance.  For example, if I run with -j20, then ctest may select 3 MPI_4 tests and 8 MPI_1 tests (3*4+8*1 = 20) in the first batch.  This makes sense, yet the reported runtime for the tests is much longer than when I run these tests individually.  I'm also not sure about the accuracy of the reported runtimes (which the timeouts would be based on).  Anyway, it all seems like a total mess.  The only reliable way to run ctest seems to be without -j.

**Comment by bartlettroscoe** [comment]
@dridzal, we can try backing off the parallel level some more but running the tests in that build is already consuming almost 2.5 hours as shown [here](https://testing.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4451880) today.

We are just going to need to disable some of these timing out tests in builds like thisnow until Kitware can implement a solution as part of #2422 (but that could be a long way off).

**Comment by dridzal** [comment]
Ok.  I just wish that there was some mechanism to periodically re-enable all the tests to see if things have improved on these platforms.  How is a developer supposed to know when to re-enable the tests?

**Comment by jmgate** [comment]
@bathmatt suggested today that I try adding `hwloc_base_binding_policy = none` to the file `~/.openmpi/mca-params.conf` to help with some similarly long test times.  Perhaps that might be of some value here?

**Comment by bartlettroscoe** [comment]
> Ok. I just wish that there was some mechanism to periodically re-enable all the tests to see if things have improved on these platforms. How is a developer supposed to know when to re-enable the tests?

@dridzal, in the (hopefully not too distant) future, we will be able to leave these tests running and mark them as "okay_to_fail=1" (don't trigger global FAIL if test does not pass) and "randomy_failing=2" (don't list them in analysis email if they pass) and then they will not impact global pass/fail for our APP customer-specific CDash analysis and reporting tool and will not clutter up our summary emails when they do pass.  That is really the best solution for tests like this I think.  We will get to that as part of #2933 in a bit.

**Comment by bartlettroscoe** [comment]
FYI: This was fixed by the PR #4702 merged to 'atdm-nightly' on 3/23/2019 and to 'develop' on  3/25/19 that improved the OpenMP settings (thanks to Brad King at Kitware).  This test is shown passing every day since 3/24/2019 in [this query](https://testing.sandia.gov/cdash/queryTests.php?project=Trilinos&filtercombine=and&filtercombine=&filtercombine=and&filtercombine=and&filtercount=4&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-cee-rhel6_intel-18.0.2_mpich2-3.2_openmp_static_opt&field2=testname&compare2=61&value2=ROL_adapters_tpetra_test_sol_TpetraSimulatedConstraintInterfaceCVaR_MPI_4&field3=site&compare3=61&value3=cee-rhel6&field4=buildstarttime&compare4=83&value4=2019-03-12T00%3A00%3A00).

Closing as resolved.
